Add AnalyticsManager.

andrew-tran-infillion opened the pull request:
Add AnalyticsManager.

  Add background location permission to Android Sample App.
  Make Android AnalyticsManager spec for old and new arch.
  Make Android AnalyticsManagerModule and add it to GimbalPackage modules.
  Make iOS AnalyticsManager.
  Add APS entitlement to iOS Sample App.
  Add AnalyticsManager test code to Sample App.
  Make NativeAnalyticsManager.ts and add it to src/index.tsx.

kimvsparrow commented on sample/android/app/src/debug/AndroidManifest.xml:
```diff
@@ -3,6 +3,7 @@
     xmlns:tools="http://schemas.android.com/tools">
 
     <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
+    <uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION" />
```
I'm pretty sure this isn't needed. This permission is already in the main manifest.  When building a debug variant, this debug manifest is applied in addition to the main manifest.

kimvsparrow commented on sample/ios/RtnGimbalSdkSample/RtnGimbalSdkSample.entitlements:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Does this do anything yet, considering that we haven't implemented `CommunicationManager`?

andrew-tran-infillion replied:
oops, this was a holdover from when Nick was asking about testing communicates, I'll remove it now and add it back in later, sounded like we're going to have more discussions surrounding CommunicationManager.

kimvsparrow commented on sample/ios/RtnGimbalSdkSample.xcworkspace/contents.xcworkspacedata:
```diff
@@ -1,6 +1,9 @@
 <Workspace
    version = "1.0">
+   <FileRef
+      location = "group:../../../../../../Documents/firestation.gpx">
```
This looks like a personal testing file, if so it can be removed.

kimvsparrow reviewed (approved):
Looks good, thanks Andrew.

LeatherGreenGargoyle committed:
Make AnalyticsManager.

Make NativeAnalyticsManager spec and add to index.ts.
Make Android AnalyticsManager spec for old and new arch.
Make Android AnalyticsManagerModule and add it to GimbalPackage modules.
Make iOS AnalyticsManager.
Add AnalyticsManager test code to Sample App.
